onnx/onnx_backend.h

```diff
@@ -0,0 +1,801 @@
+ *                                    internal error.
+ */
+ONNX_PUBLIC onnxStatus ONNX_ABI
+    ONNX_SYMBOL_NAME(onnxReleaseBackend)(onnxBackend backend);
+
+/**
+ * Parse an ONNX graph and convert it for a particular backend.
+ *
+ * Model graph is passed as a serialized ModelProto message, where types and
+ * dimensions of all inputs (including static weights) and outputs are specified
+ * through ModelProto.graph.input and ModelProto.graph.output messages. If the
+ * backend supports ONNX_CAPABILITY_SYMBOLIC_SIZE_TENSORS, some of the shape
+ * dimensions can be symbolic. In this case, their validation should be deferred
+ * until a later call to onnxSetGraphIO.
+ *
+ * Values of all static weights of the graph must be specified either in
+ * ModelProto.graph.initializer, or through the weightDescriptors parameters,
+ * but not through any combination of the two methods. If the caller creates the
+ * graph on the fly, it SHOULD pass weights through weightDescriptors as it
+ * involves less overhead.
+ *
+ * Blobs and operators in this graph are independent of the blobs and operators
+ * of other graphs on the same backend.
+ *
+ * @param backend - backend handle created by onnxInitBackend. This backend
+ *                  would be used to setup and run the model graph.
+ * @param onnxModelSize - size of the serialized ONNX ModelProto message,
+ *                        in bytes.
+ * @param[in] onnxModel - pointer to serialized ONNX ModelProto message
+ *                        representing the model graph.
+ * @param weightsCount - number of weights specified in this function call
+ *                       through tensor descriptors. Alternatively, the weights
+ *                       can be specified in ModelProto.graph.initializer
+ * @param[in] weightDescriptors - descriptors of input tensors for the graph.
+ *                                Elements of this array provide location
+ *                                for blobs identified by ValueInfoProto.name
+ *                                listed in ModelProto.graph.input of the ONNX
+ *                                graph. If this parameter is non-NULL,
+ *                                all static weights must be specified through
+ *                                the tensor descriptors, and the
+ *                                ModelProto.graph.initilizer list must be
+ *                                empty. The tensor descriptors for the weights
+ *                                must use ONNX_MEMORY_TYPE_CPU memory type,
+ *                                and the backend must copy the values of the
+ *                                weights into its own memory before the
+ *                                function returns.
+ * @param[out] graph - pointer to the opaque handle for the created ONNX graph.
+ *                     If the function fails, the handle is initialized to NULL.
+ *
+ * @retval ONNX_STATUS_SUCCESS The function call succeeded and the model graph
```

Comment:
Do we want to use something like `warn_unused_results` for these functions to notice call sites that forget to check the status?

Reply:
Yeah, thats a good idea